From: Haifeng Xu <haifeng.xu@shopee.com>
Subject: mm/mm_init.c: move set_pageblock_order() to free_area_init()
Date: Thu, 1 Jun 2023 06:35:35 +0000

pageblock_order only needs to be set once, there is no need to initialize
it in every zone/node.

Link: https://lkml.kernel.org/r/20230601063536.26882-1-haifeng.xu@shopee.com
Signed-off-by: Haifeng Xu <haifeng.xu@shopee.com>
Reviewed-by: Mike Rapoport (IBM) <rppt@kernel.org>
Cc: Michal Hocko <mhocko@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/mm_init.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/mm/mm_init.c~mm-mm_initc-move-set_pageblock_order-to-free_area_init
+++ a/mm/mm_init.c
@@ -1585,7 +1585,6 @@ static void __init free_area_init_core(s
 		if (!size)
 			continue;
 
-		set_pageblock_order();
 		setup_usemap(zone);
 		init_currently_empty_zone(zone, zone->zone_start_pfn, size);
 	}
@@ -1852,6 +1851,8 @@ void __init free_area_init(unsigned long
 	/* Initialise every node */
 	mminit_verify_pageflags_layout();
 	setup_nr_node_ids();
+	set_pageblock_order();
+
 	for_each_node(nid) {
 		pg_data_t *pgdat;
 
_
